ports/stx/portfile.cmake

```
@@ -0,0 +1,40 @@
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO lamarrr/STX
+    REF v1.0.1
+    SHA512 544ca32f07cd863082fa9688f5d56e2715b0129ff90d2a8533cc24a92c943e5848c4b2b06a71f54c12668f6e89e9e3c649f595f9eb886f671a5fa18d343f794b
+    HEAD_REF master
+    PATCHES
+      "CMakeLists.patch"
+)
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+     backtrace    STX_ENABLE_BACKTRACE
+)
+
+string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" STX_BUILD_SHARED)
+
+if(${STX_BUILD_SHARED})
+    message("STX does not currently support building a dynamic library")
+endif()

```

```suggestion
vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
```
We can use this function to take place of these codes. Please also put this line at the top of this file.

---

The issue here is not that we cannot link the STX library with the dynamic c++ runtime.
We just cannot produce a DLL file (only a static library).
This makes STX compatible with the x64-windows and x86-windows triplets but it just links statically.

---

This message does not mean the STX library cannot be used with the C++ DLL runtime.
It just cannot be built as a DLL, only a static library.
I tested it and nothing prevents a successful use of STX with the x64-windows and x86-windows triplets

---

In this case, we can also use this function.